pkg/daemon/ceph/client/mirror.go

```diff
@@ -362,6 +363,39 @@ func ListSnapshotSchedulesRecursively(context *clusterd.Context, clusterInfo *Cl
 	return snapshotSchedulesRecursive, nil
 }
 
+// EnableRBDRadosNamespaceMirroring enables rbd mirroring on a rados namespace.
+func EnableRBDRadosNamespaceMirroring(context *clusterd.Context, clusterInfo *ClusterInfo, poolAndRadosNamespaceName, remoteNamespace, mode string) error {
+	logger.Infof("enable mirroring in rados namespace %s in k8s namespace %q", poolAndRadosNamespaceName, clusterInfo.Namespace)
+	args := []string{"mirror", "pool", "enable", poolAndRadosNamespaceName, mode, "--remote-namespace", remoteNamespace}
+	cmd := NewRBDCommand(context, clusterInfo, args)
+	cmd.JsonOutput = false
+	output, err := cmd.Run()
+	if err != nil {
+		// remove the check when the min supported version is 20.0.0
+		if !clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 20, Minor: 0, Extra: 0}) {
```

Comment:
@travisn @parth-gr do you think we might need to parameterize these so they can be changed at build time for when a user's custom Ceph build has this backported?

```golang
CephBlockPoolRadosNamespaceMirroringMinimumFeatureVersionMajor := 20
CephBlockPoolRadosNamespaceMirroringMinimumFeatureVersionMinor := 0

// ...

		if !clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: CephBlockPoolRadosNamespaceMirroringMinimumFeatureVersionMajor, Minor: CephBlockPoolRadosNamespaceMirroringMinimumFeatureVersionMinor, Extra: 0}) {
```

Reply:
Good idea, trying to understand where this will be defined, makefile? As so we can change them during the `rook build`

Reply:
In a number of places we define the ceph version checks either inline (like you implemented it here) or as a package var at the top of the file. For example:
```go
MinVersionForCephExporter = cephver.CephVersion{Major: 18, Minor: 0, Extra: 0}
```
For readability I do like defining the version in a var, but this should not be a build (makefile) var.